### PR TITLE
docs: add growing your project section

### LINKED ISSUE #235: feat: Add section on Growing Your Project to the Maintainers Section
Right now, we have a Maintainers section. It includes a basic guide on how to use OpenSauced for maintainers. But what if maintainers want to grow their project? How can they do that with OpenSauced?

## Implementing the section

- [ ] Add `growing-your-project.md` to `docs > maintainers`
- [ ] In `sidebar.js`, add `maintainers/growing-your-project' path below the guide:  "[maintainers/maintainers-guide-to-open-sauced](https://github.com/open-sauced/docs/blob/a7b4fea2d1f3070d3765db64dd5b4f11a429d4f4/sidebars.js#L38C56-L38C56)",
- [ ] In `growing-your-project.md`, create the following sections:
	- [ ] When to grow your project (brief explanation of need to get more contributors or add people to teams (maintainers, triage, docs)
	- [ ] How to identify new contributors and team members
		- [ ] Look at existing contributors. How is there participation? Are they engaged? Are they invested in the success of the project?
		- [ ] Create an insight page with your projects. Add the most active contributors to a list. 	
	- [ ] Recruiting team members and contributors
		- [ ] 	Create an insight page with projects similar to yours. Add contributors you think might be a good fit to a list. After spending some time monitoring, reach out to them.  	
	- [ ] Understanding Team Participation. 
		- [ ] It's important to monitor team participation. Once you have someone on a team, add them to your list to monitor your participation. (Use the team sync feature -- link to the gif). It's useful to set a timeframe for how active they need to be.
		
		
### Additional Resources to develop this section
- https://docs.opensauced.pizza/contributing/triage-guide/
- https://docs.opensauced.pizza/features/lists/
- https://docs.opensauced.pizza/features/insights/

### PR DESCRIPTION
**Pull request by CBID2**
## Description
This PR adds a section to the Maintainers Guide on growing open source projects. It will give maintainers an idea on how OpenSauced can help them with this process. 

## Related Tickets & Documents

Closes #235 
## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [inline docs/maintainers/growing-your-project.md:45]: [question/suggestion]: Would it be fitting to mention the Pitch Your Projects X space as a tip?

**Review by CBID2 — COMMENTED** [review]
Hi @BekahHW and @adiati98! :) I finished my first draft of this guide, and added a few questions to gain some more tips on how to improve it.

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md:45]: I'm on my mobile atm and might not be able to review this today. But can you let me know to which direction you go?

Comment (CBID2) [reply docs/maintainers/growing-your-project.md:45]: Hi @adiati98. This issue was raised months ago so I based this information on the recent team workspace feature and my own experience being recruited to being a maintainer at LinksHub.

Comment (adiati98) [reply docs/maintainers/growing-your-project.md:45]: @CBID2, sorry, I didn't see that you've commented already when I edited my previous comment. 😅 
I've added an explanation about the repetitive topic that you've mentioned that you might need to read.

I have a question, though. Which recent team workspace feature do you mean?

I might be wrong. But I think what Bekah wants for this section is mainly about recruiting new team members with OpenSauced. Maintainers can use insights features as a tool to monitor contributors from other projects that have the potential to be team members. 

If this is the case, I don't see why the suggestions are pretty outdated. 🙂

That being said, I'm not sure if your suggestions fit in here. But I'll let Bekah decide.

Comment (BekahHW) [reply docs/maintainers/growing-your-project.md:48]: @CBID2 we shouldn't mention anything in the docs for events, unless it's #100Daysofoss. It's more to maintain. 

If you're thinking about going in a direction that differs from the issue, please always ask first. I think it's a good idea to talk about workspaces, but this should focus on insights.

If you have a workspace for your project(s), you can create contrib and repo insights to help you grow your project in that workspace.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:25]: We have to be careful here. Pop culture references are not accessible to everyone, and for our docs we want to make sure that a universal audience can understand and relate to them.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:29]: We have to be specific as to whether we want them to create a repository insight page or a contributor insight page.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:40]: This is not true

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md:25]: I think, unlike blog posts, docs should be something "semi-formal" if not entirely "formal". Because docs is mainly talking about products/projects, and how to interact with them. Honestly, I've never seen docs written in an informal and non-universal way anywhere. My recommendation is to write this down in "semi-formal" style like what we have now.

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md:40]: We omitted connection from [our docs](https://docs.opensauced.pizza/). So, I don't think we should even mention connections anywhere in the docs.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [reply docs/maintainers/growing-your-project.md:25]: You'll notice in that post that I don't reference specific super heroes. I'm going to push back against the idea that batman and superman are universal. 

The concept of superheroes is always more universal than specifying them by name; however, there's a difference in audience and tone with docs. I consider our docs to be semi-formal and blog posts can be more conversational, because the audience likely varies from week to week.

Docs should be consistent and relevant at all times and for the broadest audience. So we need to be very deliberate here with the style and continue to maintain the style that we've established throughout the docs.

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [reply docs/maintainers/growing-your-project.md:29]: Hey @BekahHW. Here, I decided that maintainers could use a contributor insight page. I got the inspiration from [this part of the maintainers guide](https://docs.opensauced.pizza/maintainers/maintainers-guide-to-open-sauced/#why-create-a-contributor-insight-page).

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:16]: Let's capitalize Repository and Contributor Insights page.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:24]: I don't think we need to mention devcards or even profiles here. The best way to get a quick overview is with a Contributor Insight page or by creating a Repo Insight page, looking at the contributors there and then adding them to a Contrib Insight page.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:51]: I wouldn't use crucial here. It can be "useful" or "helpful" to monitor your teams activity level by using a Contributor Insight page.

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [reply docs/maintainers/growing-your-project.md:47]: Hi @BekahHW. I made the updates! :)

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:16]: Link to docs for repo insight page. It also might be useful to have 

:::info
Repository Insights pages give an overview of open source projects, including more information on...

:::

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:7]: Let's add more key words here:
"open source maintainers" "growing an open source project" "open source project success" "maintainers guide"

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:27]:
```suggestion
When it comes to planning the growth strategies for your open source project, the people you bring on your team(s) and your project's contributors are key to its success, which is why picking the right people is important. When doing this process, consider going to the "Contributors" tab in your project's Repository Insight Page. While you do this, ask yourself the following questions:
```

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:32]: There's a grammar error and need capitalization here.

```suggestion
Once you have made your observations, [create a Contributor Insight Page for your project](../maintainers/maintainers-guide.md#creating-a-new-contributor-insight-page). This will help you to make decisions about what steps you want to take next. Here are some approaches you can use to help you:
```

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:38]: @BekahHW I have a general question here. Do we want to mention blog posts in our docs? Because I checked, so far, we don't mention any blog post in our docs.

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:46]: Add a space before and after the content of the tip to [avoid problem with Prettier](https://docusaurus.io/docs/markdown-features/admonitions#usage-with-prettier) if we have one in the future.

```suggestion
:::tip

Consider checking out our [Understanding Repository Insights Data](../features/repo-insights.md) guide to give you an idea on what to look for when narrowing down your list.

:::
```

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:42]: There are grammar errors. Also, please keep consistency for the capitalization.

```suggestion
When you're ready to recruit your team members, consider creating a [Repository Insight Page](../features/repo-insights.md) with projects that are similar to yours. From there, determine which contributors suit your project, add them to a list, and monitor their progress.  
```

**Review by adiati98 — CHANGES_REQUESTED** [review]
@CBID2 I left some reviews here.

I haven't checked the content because @BekahHW is doing that right now and I don't want to confuse any of you.
What I'm reviewing here more of the consistency purpose, nitpick and typos.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:12]: I think we can omit this first paragraph and start on line 16.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:16]: I'm not sure it makes sense to mention repo insights pages right here because it doesn't have to do with the questions you're asking. 

I would also avoid mentioning stars and forks as a goal for a project. Instead, you can ask "Are there active community members who would be interested in progressing to an official role?

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:26]: Let's make this a little clearer:

 > When it comes to planning the growth strategies for your open source project, picking the right people will determine how successful your project will be. You can discover the right people for your project by starting with a [Repository Insight Page](add link to feature). Once you've created a Repository Insight Page, you can use the Contributors dashboard to help you get a better understanding of your existing contributors.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:27]: I don't think we need this. The next line should say something like:

> If you have a reliable group of contributors, adding them to a [Contributor Insight Page](add link to feature docs) can help you to better understand their contributions to your project and the open source ecosystem.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [reply docs/maintainers/growing-your-project.md:37]: @adiati98 I think it's ok to link it in the guides, just not in the feature documentation. But I will say if there are options to link to another part of the docs, the courses, or a blog post, this is the order of prioritization:

1. docs
2. courses
3. blog posts

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:39]: I think this might be more effective if we present it in a different way. 

### Recruiting Team Members and Contributors Outside of Your Project

If you've looked at your Repository Insight Page and realize there are no contributors that could help you, you can consider recruiting new contributors. To do that, consider creating a...

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:47]: Let's remove the profile section. You can end with a simple sentence like, "Once you've identified the people you want to ask, reach out to them, providing a clear message and a list of expectations.

File: docs/maintainers/growing-your-project.md
Comment (BekahHW) [inline docs/maintainers/growing-your-project.md:52]: You have to create a team in GH to be able to use the team sync feature.

I like this heading and I think it will be a helpful paragraph, but I'd love to see you develop it more. _How_ do you understand team participation? What are the things maintainers should be considering? What happens if they see the participation is minimal?

**Review by BekahHW — CHANGES_REQUESTED** [review]
I think the base is here, but I would love to see it developed. 

Another option for this guide that would make a good heading at the end of this is, "Creating Transparency and Communication With Your Team" where you talk about creating a shared workspace to help set team goals, get insight into the project, etc.

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [inline docs/maintainers/growing-your-project.md:48]:
```suggestion
Once you've identified the people you want to ask, reach out to them by providing a clear message and a list of expectations.
```

**Comment by BekahHW** [comment]
@CBID2 just a heads-up we just merged in a pretty big update to the docs, so you'll want to pull from main and install the new dependencies.

**Comment by CBID2** [comment]
> @CBID2 just a heads-up we just merged in a pretty big update to the docs, so you'll want to pull from main and install the new dependencies.

How do I install the new dependencies @BekahHW?

**Comment by adiati98** [comment]
> > @CBID2 just a heads-up we just merged in a pretty big update to the docs, so you'll want to pull from main and install the new dependencies.
> 
> How do I install the new dependencies @BekahHW?

Just run `npm ci` as mentioned on the [README](https://github.com/open-sauced/docs?tab=readme-ov-file#%EF%B8%8F-development) after you pull the changes, @CBID2.

Then, don't forget to run it locally to test if it works.

**Comment by CBID2** [comment]
> > > @CBID2 just a heads-up we just merged in a pretty big update to the docs, so you'll want to pull from main and install the new dependencies.
> > 
> > 
> > How do I install the new dependencies @BekahHW?
> 
> Just run `npm ci` as mentioned on the [README](https://github.com/open-sauced/docs?tab=readme-ov-file#%EF%B8%8F-development) after you pull the changes, @CBID2.
> 
> Then, don't forget to run it locally to test if it works.

@adiati98, I got this error when I did the commands:

```
rror: Unsupported engine
    at checkEngine (/home/gitpod/.npm/_npx/6cea267ee22d4a04/node_modules/npm-install-checks/index.js:15:25)
    at Object.<anonymous> (/home/gitpod/.npm/_npx/6cea267ee22d4a04/node_modules/@open-sauced/check-engines/index.js:13:1)
    at Module._compile (node:internal/modules/cjs/loader:1369:14)
    at Module._extensions..js (node:internal/modules/cjs/loader:1427:10)
    at Module.load (node:internal/modules/cjs/loader:1206:32)
    at Module._load (node:internal/modules/cjs/loader:1022:12)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:135:12)
    at node:internal/main/run_main_module:28:49 {
  pkgid: '@open-sauced/docs.opensauced.pizza',
  current: { node: '20.12.2', npm: '10.5.0' },
  required: { node: '^18', npm: '>=9' },
  code: 'EBADENGINE'
}
```

**Comment by BekahHW** [comment]
@CBID2 what have you tried to fix the error?

**Comment by CBID2** [comment]
> @CBID2 what have you tried to fix the error?

Kill the terminal and do the steps again @BekahHW.

**Comment by adiati98** [comment]
@CBID2 have you tried to google the error message?

**Comment by CBID2** [comment]
> @CBID2 have you tried to google the error message?

Doing that now

**Comment by BekahHW** [comment]
> > @CBID2 have you tried to google the error message?
> 
> Doing that now

It's good practice to start by googling or asking chatGPT to help you navigate the error. Here's a Virtual Coffee [Guide](https://virtualcoffee.io/resources/developer-resources/developer-tips/asking-coding-questions#how-to-ask-coding-questions) that might help you develop a process for navigating errors.

**Comment by CBID2** [comment]
> > > @CBID2 have you tried to google the error message?
> > 
> > 
> > Doing that now
> 
> It's good practice to start by googling or asking chatGPT to help you navigate the error. Here's a Virtual Coffee [Guide](https://virtualcoffee.io/resources/developer-resources/developer-tips/asking-coding-questions#how-to-ask-coding-questions) that might help you develop a process for navigating errors.

I got it to work @BekahHW and @adiati98! :) It turns out I had to downgrade my version of Nodejs

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:56]: - What is "Act"?
- It needs a period at the end of the sentence.
- Inconsistency. You use  the term "section" in the other bullet points, and "tab" here.

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:66]:
```suggestion
The methods utilized to collaborate and communicate with your team play a crucial role in the growth of your open source project. Creating and sharing your project's [Workspace](../features/workspaces.md) page with your team members can be helpful in fostering a collaborative environment. After creating this page, here are some ways to leverage its information to engage with your team:
```

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:68]: I don't quite understand what do you mean here. Can you help me to understand it better?

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [inline docs/maintainers/growing-your-project.md:69]: There is no Engagement "tab".

**Review by adiati98 — CHANGES_REQUESTED** [review]
Hi @CBID2,

I left some reviews here. :)

I would like to give you a suggestion to read the docs, especially those are related to what you write here.
And you might want to explore the app so you can relate what are written in the docs and how the features actually work.

Don't hesitate to ask us if you have questions!

Also, while you're waiting for Bekah's review regarding to the content, you can spend time to reread your changes and see if you can fix or improve anything.

Thank you!

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [reply docs/maintainers/growing-your-project.md:69]: Yes there is @adiati98 

screenshot showing Engagement tab

File: docs/maintainers/growing-your-project.md
Comment (CBID2) [reply docs/maintainers/growing-your-project.md:68]: I was referring to the dashboard in the Workspace page(refer to the red symbol in the Engagement tab in the screenshot below)
screenshot of engagement tab

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md:69]: I know there's an "Engagement", but I'm not sure if it calls a "tab".

Comment (CBID2) [reply docs/maintainers/growing-your-project.md:69]: Ah so that's the name! I think statistic card would be a good name.

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md]: Ah, I see. You're looking at those.

However, you might want to formulate the wording better. I'm not sure about native-English speakers, but I certainly don't understand this at all. 😅

File: docs/maintainers/growing-your-project.md
Comment (adiati98) [reply docs/maintainers/growing-your-project.md]: Yep. Also, it would be great if you try to search around first, then ask for clarity with some explanation of what you did and you've get so far when you're not sure about something rather than assuming. 🙂

**Comment by BekahHW** [comment]
@CBID2 when you feel like things are ready, please mark the PR as "Ready for Review"

**Comment by CBID2** [comment]
> @CBID2 when you feel like things are ready, please mark the PR as "Ready for Review"

I'm ready now @BekahHW.

**Comment by CBID2** [comment]
Hey @BekahHW and @adiati98. Should I add information about the StarSearch feature? I've been tinkering around with it and found some ways it can be applicable to this guide.

**Comment by adiati98** [comment]
@CBID2,
Have you run and test this locally?
Because when I run this locally, I see this:

Screenshot 2024-05-27 164933

**Comment by CBID2** [comment]
> @CBID2, Have you run and test this locally? Because when I run this locally, I see this:
> 
> Screenshot 2024-05-27 164933

Hey @adiati98. I made the change and ran it locally. Here's how it looks
screenshot of note

**Comment by BekahHW** [comment]
Hey @CBID2, 

We noticed that you have some unverified commits in your PR.

Can you look into what's going on with this? Here's the GH docs on [verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).

image

**Comment by BekahHW** [comment]
> > Hey @CBID2,
> > We noticed that you have some unverified commits in your PR.
> > Can you look into what's going on with this? Here's the GH docs on [verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).
> > image
> 
> Hey @BekahHW. I have been switching between using VS.Code's Source Control and the terminal to push my commits.

Looks like you might need to set it up in one of them.

**Comment by BekahHW** [comment]
Hey @CBID2, I converted this to a draft until you have some time to go through the comments and create a revision strategy.

**Comment by CBID2** [comment]
> Hey @CBID2, I converted this to a draft until you have some time to go through the comments and create a revision strategy.

Hi @BekahHW. I created a new PR for this issue. It's #324

**Pull request closed without merging**
